policy/scan/local_scanner.go

```diff
@@ -462,6 +468,31 @@ func (s *LocalScanner) HealthCheck(ctx context.Context, req *HealthCheckRequest)
 	}, nil
 }
 
+func (s *LocalScanner) getUpstreamConfig(incognito bool, job *Job) resources.UpstreamConfig {
+	if incognito {
+		return resources.UpstreamConfig{Incognito: true}
+	}
+
+	plugins := s.plugins
+	endpoint := s.apiEndpoint
+	spaceMrn := s.spaceMrn
+
+	jobCredentials := job.Inventory.Spec.UpstreamCredentals
+	if s.allowJobCredentials && jobCredentials != nil {
+		certAuth, _ := upstream.NewServiceAccountRangerPlugin(jobCredentials)
+		plugins = append(plugins, certAuth)
```

Comment:
I think this is problematic. If we have already and authentication plugin attached we add just another one. We need to split the authentication plugin from the rest of the plugins

Reply:
Internally, ranger will always prefer the latest one as it builds out a map. However, I changed the plugins list to a map in the local scanner so that we can be very explicit about which header should be in the request